app/src/sharedTest/java/org/oppia/android/app/databinding/MarginBindableAdaptersTest.kt

```diff
@@ -0,0 +1,341 @@
+    val textView = activityRule.scenario.runWithActivity {
+      val textView: TextView = it.findViewById(R.id.test_margin_text_view)
+      setLayoutMarginStartAndMarginEnd(textView)
+      return@runWithActivity textView
+    }
+    assertThat(textView.marginStart.toFloat()).isWithin(TOLERANCE).of(
+      convertToPx(24f)
+    )
+    assertThat(textView.marginEnd.toFloat()).isWithin(TOLERANCE).of(
+      convertToPx(40f)
+    )
+  }
+
+  @Config(qualifiers = "land")
+  @Test
+  fun testMarginBindableAdapters_ltrIsEnabled_landscape_marginStartAndMarginEndForLtrIsCorrect() {
+    val textView = activityRule.scenario.runWithActivity {
+      val textView: TextView = it.findViewById(R.id.test_margin_text_view)
+      setLayoutMarginStartAndMarginEnd(textView)
+      return@runWithActivity textView
+    }
+    assertThat(textView.marginStart.toFloat()).isWithin(TOLERANCE).of(
+      convertToPx(24f)
+    )
+    assertThat(textView.marginEnd.toFloat()).isWithin(TOLERANCE).of(
+      convertToPx(40f)
+    )
+  }
+
+  @Config(qualifiers = "sw600dp-port")
+  @Test
+  fun testMarginBindableAdapters_ltrEnabled_port_tablet_marginStartAndMarginEndForLtrIsCorrect() {
+    val textView = activityRule.scenario.runWithActivity {
+      val textView: TextView = it.findViewById(R.id.test_margin_text_view)
+      setLayoutMarginStartAndMarginEnd(textView)
+      return@runWithActivity textView
+    }
+    assertThat(textView.marginStart.toFloat()).isWithin(TOLERANCE).of(
+      convertToPx(24f)
+    )
+    assertThat(textView.marginEnd.toFloat()).isWithin(TOLERANCE).of(
+      convertToPx(40f)
+    )
+  }
+
+  @Config(qualifiers = "sw600dp-land")
+  @Test
+  fun testMarginBindableAdapters_ltrEnabled_landscapeTablet_marginStartAndEndForLtrIsCorrect() {
+    val textView = activityRule.scenario.runWithActivity {
+      val textView: TextView = it.findViewById(R.id.test_margin_text_view)
+      setLayoutMarginStartAndMarginEnd(textView)
+      return@runWithActivity textView
+    }
+    assertThat(textView.marginStart.toFloat()).isWithin(TOLERANCE).of(
+      convertToPx(24f)
+    )
+    assertThat(textView.marginEnd.toFloat()).isWithin(TOLERANCE).of(
+      convertToPx(40f)
+    )
+  }
+
+  @Config(qualifiers = "port")
+  @Test
+  fun testMarginBindableAdapters_rtlIsEnabled_port_marginStartAndMarginEndForRtlIsCorrect() {
+    val textView = activityRule.scenario.runWithActivity {
+      val textView: TextView = it.findViewById(R.id.test_margin_text_view)
+      setLayoutMarginStartAndMarginEnd(textView)
+      textView.layoutDirection = View.LAYOUT_DIRECTION_RTL
+      return@runWithActivity textView
+    }
+    assertThat(textView.marginEnd.toFloat()).isWithin(TOLERANCE).of(
+      convertToPx(24f)
+    )
+    assertThat(textView.marginStart.toFloat()).isWithin(TOLERANCE).of(
+      convertToPx(40f)
+    )
+  }
+
+  @Test
+  fun testMarginBindableAdapters_rtlIsEnabled_port_marginStartAndMarginEndForRtlFails() {
```

Comment:
Test case where RTL start and end margin fails.

Reply:
Isn't this test backwards? We're asserting that the start/end are staying the same rather than flipping.

Reply:
This test I added as example to show how it is working. I will be removing this.